crates/matrix-sdk/src/error.rs

```diff
@@ -422,3 +423,47 @@ pub enum RefreshTokenError {
     #[error("the access token could not be refreshed")]
     UnableToRefreshToken,
 }
+
+/// Errors that can occur when manipulating push notification settings.
+#[derive(Debug, Error, Clone)]
+pub enum NotificationSettingsError {
+    /// Invalid room id.
+    #[error("Invalid room id")]
+    InvalidRoomId,
```

Comment:
I removed this error because the `room_id` is now an instance of `RoomId` and so this case will no longer happen in the sdk. As for the ffi part, I've added the room id to the error as suggested here.